rust-multiaddr/src/lib.rs

```diff
@@ -0,0 +1,375 @@
+///! # multiaddr
+///!
+///! Implementation of [multiaddr](https://github.com/jbenet/multiaddr)
+///! in Rust.
+extern crate byteorder;
+extern crate cid;
+extern crate integer_encoding;
+
+mod protocol;
+mod errors;
+
+pub use errors::{Result, Error};
+pub use protocol::{ProtocolId, ProtocolArgSize, AddrComponent};
+
+use std::fmt;
+use std::iter::FromIterator;
+use std::net::{SocketAddr, SocketAddrV4, SocketAddrV6, IpAddr, Ipv4Addr, Ipv6Addr};
+use std::str::FromStr;
+
+/// Representation of a Multiaddr.
+#[derive(PartialEq, Eq, Clone, Hash)]
+pub struct Multiaddr {
+    bytes: Vec<u8>,
+}
+
+impl ToString for Multiaddr {
+    /// Convert a Multiaddr to a string
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use multiaddr::Multiaddr;
+    ///
+    /// let address = Multiaddr::new("/ip4/127.0.0.1/udt").unwrap();
+    /// assert_eq!(address.to_string(), "/ip4/127.0.0.1/udt");
+    /// ```
+    ///
+    fn to_string(&self) -> String {
+        let mut out = String::new();
+        for s in self.iter() {
+            out.push_str(&s.to_string());
+        }
+        out
+    }
+}
+
+impl fmt::Debug for Multiaddr {
```

Comment:
Can we implement `Display` too and have `to_string` just be `format!("{}", self)`?

Reply:
@tomaka Can you either address this or explain why you don't want to address it?

Reply:
Done ; note that `ToString` is automatically implemented on `T where T: Display` by the stdlib.